jujugui/static/gui/src/app/components/header-logo/header-logo.js

```diff
@@ -0,0 +1,59 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2017 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+/**
+  We want the header logo to be a component as we may need to change what
+  happens when it's clicked.
+*/
+YUI.add('header-logo', function() {
+
+  juju.components.HeaderLogo = React.createClass({
+
+    propTypes: {
+      changeState: React.PropTypes.func.isRequired,
+      user: React.PropTypes.object.isRequired
+    },
+
+    /**
+      Change state to the user profile when clicked for when you're browsing
+      another user/model
+
+     @method _handleLogoClick
+    */
+    _handleLogoClick: function() {
+      this.props.changeState({
+        user: this.props.user
```

Comment:
This is not right, we should set the profile key, no need to disambiguate the user here. See above.
AT that point, this method is not even required, as you could have `<a onClick={this.props.navigateUserProfile}` below.